src/api/api.go

```diff
@@ -39,7 +39,7 @@ func New(conf Config) (http.Handler, error) {
 	authJwtIssuer := auth.NewJwtIssuer("treenq-api", []byte(conf.AuthPrivateKey), []byte(conf.AuthPublicKey), conf.AuthTtl)
 	githubClient := repo.NewGithubClient(githubJwtIssuer, http.DefaultClient)
 	gitDir := filepath.Join(wd, "gits")
-	gitClient := repo.NewGit(gitDir)
+	gitClient := repo.NewGit(gitDir, progressBuf)
```

Comment:
here git has a different write that' not shared with the handler and docker artifactory.
if you want to inject we must do it for the others components too, or just use a global instance and pass as a Clone call argument.

I did it a global one by purpose, it doesn't persist data and must be changed to another data source or sync-like mechanic